### ISSUE TITLE
Invalid option in build call buildConcurrency

### ISSUE THREAD

**Issue by kdybicz** [serverless/serverless#12616]
### Issue description

Based on the documentation found [here](https://www.serverless.com/framework/docs/providers/aws/guide/building) I was trying to fine-tune the build concurrency with such config:
```
build:
  esbuild:
    bundle: true
    minify: false
    sourcemap: false
    buildConcurrency: 3
```

When running `npx sls package --stage staging` I got:
```
Packaging "phoenix" for stage "staging" (eu-west-1)

⠋ Packaging✘ [ERROR] Invalid option in build() call: "buildConcurrency"

    /Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:253:12:
      253 │       throw new Error(`Invalid option ${where}: ${quote(key)}`);
          ╵             ^

    at checkForInvalidFlags (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:253:13)
    at flagsForBuildOptions (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:426:3)
    at buildOrContextContinue (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:913:9)
    at buildOrContextImpl (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:897:5)
    at Object.buildOrContext (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:699:5)
    at /Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:2023:15
    at new Promise (<anonymous>)
    at Object.build (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:2022:25)
    at Module.build (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:1873:51)
    at file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:10537
    at file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:3191
    at run4 (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:3216)
    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)
    at bound (node:async_hooks:238:16)
    at file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:3482

✘ [ERROR] Invalid option in build() call: "buildConcurrency"

    /Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:253:12:
      253 │       throw new Error(`Invalid option ${where}: ${quote(key)}`);
          ╵             ^

    at checkForInvalidFlags (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:253:13)
    at flagsForBuildOptions (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:426:3)
    at buildOrContextContinue (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:913:9)
    at buildOrContextImpl (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:897:5)
    at Object.buildOrContext (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:699:5)
    at /Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:2023:15
    at new Promise (<anonymous>)
    at Object.build (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:2022:25)
    at Module.build (/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:1873:51)
    at file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:10537
    at file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:3191
    at run4 (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:3216)
    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)
    at bound (node:async_hooks:238:16)
    at file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:3482


✖ Build failed with 1 error:
/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:253:12: ERROR: Invalid option in build() call: "buildConcurrency"
ServerlessError2: Build failed with 1 error:
/Users/kamil/.serverless/releases/4.1.5/package/node_modules/esbuild/lib/main.js:253:12: ERROR: Invalid option in build() call: "buildConcurrency"
    at Esbuild._build (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:11533)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async before:package:createDeploymentArtifacts (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1026:4371)
    at async PluginManager.runHooks (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1028:9294)
    at async PluginManager.invoke (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1028:10005)
    at async PluginManager.run (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1028:10788)
    at async Serverless.run (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1035:10442)
    at async runFramework (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1079:2314)
    at async route (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1088:14692)
    at async Object.run2 [as run] (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1088:15090)
    at async run3 (file:///Users/kamil/.serverless/releases/4.1.5/package/dist/sf-core.js:1088:16324)

For help, try the following:
  • Run the command again with the "--debug" option
  • Run "serverless support"
  • Review the docs: https://www.serverless.com/framework/docs/
```

### Context

# Service Overview
- Serverless Framework Version: 4.1.5
- Service Path: /Users/kamil/work/Phoenix
- Service Config File: serverless.yml
- Service Name: phoenix
- Service Runtime: nodejs20.x
- Service Stage: staging
- Service Region: eu-west-1
- Command: package --stage staging
- Error Code: ESBULD_BUILD_ERROR

**Comment by austencollins** [comment]
Thanks for reporting @kdybicz Looking into this..

**Comment by eahefnawy** [comment]
Thanks for reporting @kdybicz ... This was fixed in https://github.com/serverless/serverless/pull/12646 and will be included in the next release 🙌

**Comment by Mmarzex** [comment]
This has been released with [4.1.7](https://github.com/serverless/serverless/releases/tag/v4.1.7)
